rest_framework/serializers.py

```diff
@@ -562,7 +562,7 @@ def to_representation(self, data):
         """
         # Dealing with nested relationships, data can be a Manager,
         # so, first get a queryset from the Manager if needed
-        iterable = data.all() if isinstance(data, models.Manager) else data
+        iterable = data.all() if hasattr(data, 'all') else data
         return [
```

Comment:
Actually I think `isinstance(data, (models.Manager, models.QuerySet))` may be what we want.
We do need it for manager (as it's not an iterable by itself) and we also need it for queryset (as they get cached).